src/neural/encoder.cc

```diff
@@ -57,11 +58,16 @@ InputPlanes EncodePositionForNN(const PositionHistory& history,
   bool flip = false;
   int history_idx = history.GetLength() - 1;
   for (int i = 0; i < std::min(history_planes, kMoveHistory);
-       ++i, flip = !flip, --history_idx) {
-    if (history_idx < 0) break;
-    const Position& position = history.GetPositionAt(history_idx);
+       ++i, --history_idx) {
+    const Position& position =
+        history.GetPositionAt(history_idx < 0 ? 0 : history_idx);
     const ChessBoard& board =
         flip ? position.GetThemBoard() : position.GetBoard();
+    if (history_idx < 0 && fill_empty_history == NO) break;
+    if (history_idx < 0 && fill_empty_history == FEN_ONLY &&
+        (board.ours() + board.theirs()).as_int() == 0xFFFF00000000FFFFULL) {
```

Comment:
In tests of chess960 positions, Leela shown that it didn't play optimally without history (e.g. played e4 when the best move would be `1. Ne3`). I think chess960 positions should be counted as "fen".
As well as different exotic positions like "knightmare", "startpos without castling rights", etc.
Or we can just enable it for all positions, and not distinguish fen from non-fen.